python/ray/data/dataset.py

```diff
@@ -352,6 +352,7 @@ def repartition(self, num_blocks: int, *,
 
         # Compute the (n-1) indices needed for an equal split of the data.
         count = self.count()
+        num_blocks = min(num_blocks, count)
```

Comment:
Why do we need this? Does this violate the public doc `Repartition the dataset into exactly this number of blocks.`?

Reply:
This was a driveby fixing the case when repartitioning into more blocks than there are rows (yielding empty blocks). Namely, empty simple blocks are incorrectly coerced to Arrow blocks downstream.

You bring up a good point, the better solution here is to fix the downstream coercion and to keep the existing semantics of getting back exactly `num_blocks` blocks with some possibly being empty.